docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -1,763 +1,166 @@
-
 </Step>
 
-<Step label="6">
-
-### Run tests on EAS Build
-
-Running the tests on EAS Build is like running a regular build:
-
-<Terminal cmd={['$ eas build -p all -e test']} />
-
-If you have set up everything correctly you should see the successful test result in the build logs:
-
-<ContentSpotlight src="/static/images/eas-build/tests/03-logs.png" className="max-w-[720px]" />
-
-</Step>
-
-<Step label="7">
-### Upload screenshots of failed test cases
-
-> This step is optional but highly recommended.
-
-When an E2E test case fails, it can be helpful to see the screenshot of the application state. EAS Build makes it easy to upload any arbitrary build artifacts using [`buildArtifactPaths`](/eas/json/#buildartifactpaths) in **eas.json**.
-
-#### Take screenshots for failed tests
-
-Detox supports taking in-test screenshots of the device. The [**.detoxrc.js** sample](/build-reference/e2e-tests/#configure-detox) above includes a configuration for Detox to take screenshots of failed tests.
+<Step label="4">
+### Create custom build workflow for running Maestro E2E tests
+
+The easiest way to run Maestro E2E tests on EAS Build is to create a [custom build workflow](/custom-builds/get-started/).
+This workflow will build your app and run the Maestro tests on it.
+
+To do it you must add a custom build config file in your project. Create a directory **.eas/build** at the same level as **eas.json**.
+For example, if the **eas.json** file is at the root of your project, create the directory at that level.
+The path and the name of both directories are important for EAS Build to identify that a project contains a custom build config.
+
+Inside it, you can create a new config file called **build-and-maestro-test.yml**.
+This file defines the custom build workflow config that you want to run.
+Workflow contains steps that are executed during the custom build process.
+This custom build config will execute the [`eas/build`](custom-builds/schema/#easbuild) custom function group to create a build and then the [`eas/maestro_test`](custom-builds/schema/#easmaestro_test) all-in-one custom function group that installs Maestro, prepares a testing environment (Android Emulator or iOS Simulator) and tests the app using flows specified by the `flow_path` input.
+
+```yaml .eas/build/build-and-maestro-test.yml
+build:
+  name: Create a build and run Maestro tests on it
+  steps:
+    - eas/build
+    - eas/maestro_test:
+        inputs:
+          flow_path: |
+            maestro/home.yaml
+            maestro/expand_test.yaml
+```
 
-#### Configure EAS Build for screenshots upload
+Now let's modify your **eas.json** by adding a new [build profile](/build/eas-json/#build-profiles) called `build-and-maestro-test` that will be used to run the custom build config from the **build-and-maestro-test.yml** file.
+This configuration will build the simulator/emulator version of your app and run the Maestro tests on it.
```

Comment:
```suggestion
Now modify the **eas.json** by adding a new [build profile](/build/eas-json/#build-profiles) called `build-and-maestro-test`. It will be used to run the custom build config from the **build-and-maestro-test.yml** file. This configuration will build the emulator/simulator version of your app and run the Maestro tests on it.
```